src/interpreter/Environment.ts

```diff
@@ -174,11 +190,30 @@ export class Environment {
         newEnvironment.global = this.global;
         newEnvironment.module = this.module;
         newEnvironment.mPointer = this.mPointer;
+        newEnvironment.mock = this.mock;
+        newEnvironment.mockObjects = this.mockObjects;
         newEnvironment.focusedNode = this.focusedNode;
 
         return newEnvironment;
     }
 
+    /**
+     * retrieves mocked object if it exists
+     * @param objName the object to mock
+     */
+    public getMock(objName: string): BrsType {
+        return this.mockObjects.get(objName) || BrsInvalid.Instance;
+    }
+
+    /**
+     * places the mockValue object into list of mocks
+     * @param objName the object we are mocking
+     * @param mockValue the mock to return
+     */
+    public setMock(objName: string, mockValue: BrsType): void {
+        this.mockObjects.set(objName.toLowerCase(), mockValue);
```

Comment:
if we're saving the one reference of mockValue object, wouldn't its value get changed anytime someone modifies it?  so next time create mock is called, the object is no longer the original mock value